jvb/src/main/kotlin/org/jitsi/videobridge/EndpointK.kt

```diff
@@ -0,0 +1,490 @@
+                logger.info("DTLS handshake complete")
+                _transceiver.setSrtpInformation(chosenSrtpProtectionProfile, tlsRole, keyingMaterial)
+                // TODO(brian): the old code would work even if the sctp connection was created after
+                //  the handshake had completed, but this won't (since this is a one-time event).  do
+                //  we need to worry about that case?
+                sctpSocket.ifPresent(::acceptSctpConnection)
+                scheduleEndpointMessageTransportTimeout()
+            }
+        }
+    }
+
+    override fun updateForceMute() {
+        var audioForceMuted = false
+        var videoForceMuted = false
+        channelShims.forEach { channelShim ->
+            if (!channelShim.allowIncomingMedia()) {
+                when (channelShim.mediaType) {
+                    MediaType.AUDIO -> audioForceMuted = true
+                    MediaType.VIDEO -> videoForceMuted = true
+                    else -> Unit
+                }
+            }
+        }
+
+        _transceiver.forceMuteAudio(audioForceMuted)
+        _transceiver.forceMuteVideo(videoForceMuted)
+    }
+
+    override fun addPayloadType(payloadType: PayloadType) {
+        _transceiver.addPayloadType(payloadType)
+        bitrateController.addPayloadType(payloadType)
+    }
+
+    override fun addRtpExtension(rtpExtension: RtpExtension) {
+        _transceiver.addRtpExtension(rtpExtension)
+    }
+
+    override fun addReceiveSsrc(ssrc: Long, mediaType: MediaType) {
+        logger.cdebug { "Adding receive ssrc $ssrc of type $mediaType" }
+        _transceiver.addReceiveSsrc(ssrc, mediaType)
+    }
+
+    override fun onNewSsrcAssociation(
+        endpointId: String,
+        primarySsrc: Long,
+        secondarySsrc: Long,
+        type: SsrcAssociationType
+    ) {
+        if (endpointId.equals(id, ignoreCase = true)) {
+            _transceiver.addSsrcAssociation(LocalSsrcAssociation(primarySsrc, secondarySsrc, type))
+        } else {
+            _transceiver.addSsrcAssociation(RemoteSsrcAssociation(primarySsrc, secondarySsrc, type))
+        }
+    }
+
+    override fun setFeature(feature: EndpointDebugFeatures, enabled: Boolean) {
+        when (feature) {
+            EndpointDebugFeatures.PCAP_DUMP -> _transceiver.setFeature(Features.TRANSCEIVER_PCAP_DUMP, enabled)
+        }
+    }
+
+    override fun isFeatureEnabled(feature: EndpointDebugFeatures): Boolean {
+        return when (feature) {
+            EndpointDebugFeatures.PCAP_DUMP -> _transceiver.isFeatureEnabled(Features.TRANSCEIVER_PCAP_DUMP)
+        }
+    }
+
+    override fun isSendingAudio(): Boolean {
+        // The endpoint is sending audio if we (the transceiver) are receiving audio.
+        return _transceiver.isReceivingAudio()
+    }
+    override fun isSendingVideo(): Boolean {
+        // The endpoint is sending video if we (the transceiver) are receiving video.
+        return _transceiver.isReceivingAudio()
```

Comment:
Copy & paste error - this should be `isReceivingVideo()`.

Reply:
😅  nice catch